zuul-core/src/main/java/com/netflix/zuul/netty/server/Server.java

```diff
@@ -147,47 +188,40 @@ public void gracefullyShutdownConnections()
         clientConnectionsShutdown.gracefullyShutdownClientChannels();
     }
 
-    private ChannelFuture setupServerBootstrap(int port, ChannelInitializer channelInitializer)
-            throws InterruptedException
-    {
-        ServerBootstrap serverBootstrap = new ServerBootstrap().group(
-                serverGroup.clientToProxyBossPool,
-                serverGroup.clientToProxyWorkerPool);
+    private ChannelFuture setupServerBootstrap(
+            SocketAddress listenAddress, ChannelInitializer<?> channelInitializer) throws InterruptedException {
+        ServerBootstrap serverBootstrap =
+                new ServerBootstrap().group(serverGroup.clientToProxyBossPool, serverGroup.clientToProxyWorkerPool);
 
         // Choose socket options.
         Map<ChannelOption, Object> channelOptions = new HashMap<>();
         channelOptions.put(ChannelOption.SO_BACKLOG, 128);
-        //channelOptions.put(ChannelOption.SO_TIMEOUT, SERVER_SOCKET_TIMEOUT.get());
         channelOptions.put(ChannelOption.SO_LINGER, -1);
         channelOptions.put(ChannelOption.TCP_NODELAY, true);
         channelOptions.put(ChannelOption.SO_KEEPALIVE, true);
 
-        // Choose EPoll or NIO.
-        if (USE_EPOLL.get()) {
-            LOG.warn("Proxy listening with TCP transport using EPOLL");
-            serverBootstrap = serverBootstrap.channel(EpollServerSocketChannel.class);
-            channelOptions.put(EpollChannelOption.TCP_DEFER_ACCEPT, Integer.valueOf(-1));
-        }
-        else {
-            LOG.warn("Proxy listening with TCP transport using NIO");
-            serverBootstrap = serverBootstrap.channel(NioServerSocketChannel.class);
-        }
+        LOG.debug("Proxy listening with " + serverGroup.channelType);
+        serverBootstrap.channel(serverGroup.channelType);
 
         // Apply socket options.
         for (Map.Entry<ChannelOption, Object> optionEntry : channelOptions.entrySet()) {
             serverBootstrap = serverBootstrap.option(optionEntry.getKey(), optionEntry.getValue());
         }
+        // Apply transport specific socket options.
+        for (Map.Entry<ChannelOption, ?> optionEntry : serverGroup.transportChannelOptions.entrySet()) {
+            serverBootstrap = serverBootstrap.option(optionEntry.getKey(), optionEntry.getValue());
+        }
 
         serverBootstrap.childHandler(channelInitializer);
         serverBootstrap.validate();
 
-        LOG.info("Binding to port: " + port);
+        LOG.debug("Binding to : " + listenAddress);
```

Comment:
This is actually a really useful log statement when you have multiple ports being bound.

Reply:
That may be, but again, the caller of this code could log this.    This class is more of a library, and probably should let a higher layer log.

Reply:
Done